js/main.js

```diff
@@ -142,7 +142,7 @@ class App extends React.Component {
     }
 
     var url = window.location;
-    var socket = new WebSocket('ws://' + url.host + '/socket');
+    var socket = new WebSocket('ws://' + url.host + url.pathname + 'socket');
```

Comment:
window.location.pathname doesn't always end in a `/`, will that be a problem here?

Reply:
Hi,
I am not familiar with js, from my tests in this code, `url.pathname` contains either a single `/` or `/some_things/`.
I am not sure where to look online to find exactly what this should contain, are there cases I'm missing?

Reply:
Ah that would make some sense based on our current codebase, however url.pathname doesn't always append that final slash (it might return `/pathname`) so this might break in the future or if somebody is trying to make their own changes to visdom. Rather than have them need to fix it then I'd rather fix it in advance.

Can you instead declare a variable `var pathname = url.pathname` and then use a conditional `if pathname.slice(-1) != '/' { pathname = pathname + '/'`? This could also be a separate function `correctPathname` so you don't need to put it in both locations

Sorry for being somewhat picky on an otherwise great change to have!

Reply:
No problem !

One last question: can `windown.location.pathname` be an empty string? Or will it at least be `'/'` ?